gpustack/server/controllers.py

```diff
@@ -821,38 +883,58 @@ async def sync_main_model_file_state(
     )
 
     need_update = False
-    if (
-        file.state == ModelFileStateEnum.DOWNLOADING
-        and instance.state == ModelInstanceStateEnum.INITIALIZING
-    ):
-        # Download started
-        instance.state = ModelInstanceStateEnum.DOWNLOADING
-        instance.download_progress = 0
-        instance.state_message = ""
-        need_update = True
-    elif (
-        file.state == ModelFileStateEnum.DOWNLOADING
-        and instance.state == ModelInstanceStateEnum.DOWNLOADING
-        and file.download_progress != instance.download_progress
-    ):
-        # Update the download progress
-        instance.download_progress = file.download_progress
-        need_update = True
 
+    # Downloading
+    if file.state == ModelFileStateEnum.DOWNLOADING:
+        if instance.state == ModelInstanceStateEnum.INITIALIZING:
+            # Download started
+            instance.state = ModelInstanceStateEnum.DOWNLOADING
+            instance.download_progress = 0
+            instance.state_message = ""
+            need_update = True
```

Comment:
Maybe we can use this `if session.is_modified(instance, include_collections=False)` to replace need_update field.

Reply:
It's not ideal for this use case, it's more expensive and may not reflect expected state. See the instruction and caveats of the function.